substrate/frame/staking/src/lib.rs

```diff
@@ -1335,6 +1335,24 @@ impl<T: Config> EraInfo<T> {
 	}
 }
 
+/// A utility struct that provides a way to check if a given account is a staker.
+///
+/// This struct implements the `Contains` trait, allowing it to determine whether
+/// a particular account is currently staking by checking if the account exists in
+/// the staking ledger.
+pub struct AllStakers<T: Config>(core::marker::PhantomData<T>);
```

Comment:
I think I would name this something like `AllDirectStakers` or `AllNativeStakers`, since people who join nomination pools could still be considered "stakers" (i.e. they have their funds at stake). This is checking that they have an entry in the staking ledger specific to their account.